lib/protocol/message.go

```diff
@@ -159,6 +167,13 @@ type Device struct {
 	Options         []Option // max:64
 }
 
+type FileDownloadProgressUpdate struct {
+	Name         string // max:8192
+	Version      Vector
+	BlockIndexes []int32 // max:1000000
+	UpdateType   uint32
```

Comment:
I suggest moving this field to the top, as it kind of gives context to the whole thing and it feels sad to tuck it in the end. Not for any technical reason though.